csv/src/fs2/data/csv/CellDecoder.scala

```diff
@@ -53,25 +95,94 @@ object CellDecoder {
   implicit val doubleDecoder: CellDecoder[Double] = s =>
-    Try(s.toDouble).toEither.leftMap(new DecoderError(s"unable to decode '$s' as a double", _))
+    Either.catchNonFatal(s.toDouble).leftMap(new DecoderError(s"unable to decode '$s' as a double", _))
+  implicit val bigDecimalDecoder: CellDecoder[BigDecimal] = s =>
+    Try(BigDecimal(s)).toEither.leftMap(new DecoderError(s"unable to decode '$s' as a BigDecimal", _))
+  implicit val bigIntDecoder: CellDecoder[BigInt] = s =>
+    Try(BigInt(s)).toEither.leftMap(new DecoderError(s"unable to decode '$s' as a BigInt", _))
   implicit val stringDecoder: CellDecoder[String] = s => Right(s)
+  implicit val charArrayDecoder: CellDecoder[Array[Char]] = s => Right(s.toCharArray)
+
+  // Containers
+  implicit def decoderResultDecoder[T](implicit ev: CellDecoder[T]): CellDecoder[DecoderResult[T]] = s =>
+    ev(s).asRight
+  implicit def rawOrResultDecoder[T](implicit ev: CellDecoder[T]): CellDecoder[Either[String, T]] = s =>
+    ev(s).leftMap(_ => s).asRight
+  implicit def optionDecoder[T](implicit ev: CellDecoder[T]): CellDecoder[Option[T]] = s =>
+    if (s.isEmpty)
+      Right(None)
+    else
+      ev(s).map(Some(_))
+
+  // Standard Library types
+  implicit val finiteDurationDecoder: CellDecoder[FiniteDuration] =
+    durationDecoder.emap {
+      case fd: FiniteDuration => fd.asRight
+      case d => new DecoderError(s"Parsed duration isn't finite: $d").asLeft
+    }
+
+  implicit val javaUrlDecoder: CellDecoder[URL] = s =>
+    Either.catchOnly[MalformedURLException](new URL(s))
+      .leftMap(t => new DecoderError(s"Couldn't parse URL: ${t.getMessage}", t))
+
+  implicit val uuidDecoder: CellDecoder[UUID] = s =>
+    Either.catchOnly[IllegalArgumentException](UUID.fromString(s))
+      .leftMap(t => new DecoderError(s"Couldn't parse UUID: ${t.getMessage}", t))
+
+  // Java Time
+  implicit val instantDecoder: CellDecoder[Instant] = javaTimeDecoder("Instant", Instant.parse)
+  implicit val periodDecoder: CellDecoder[Period] = javaTimeDecoder("Period", Period.parse)
+  implicit val localDateDecoder: CellDecoder[LocalDate] = javaTimeDecoder("LocalDate", LocalDate.parse)
+  implicit val localDateTimeDecoder: CellDecoder[LocalDateTime] = javaTimeDecoder("LocalDateTime", LocalDateTime.parse)
+  implicit val localTimeDecoder: CellDecoder[LocalTime] = javaTimeDecoder("LocalTime", LocalTime.parse)
+  implicit val offsetDateTimeDecoder: CellDecoder[OffsetDateTime] = javaTimeDecoder("OffsetDateTime", OffsetDateTime.parse)
+  implicit val offsetTimeDecoder: CellDecoder[OffsetTime] = javaTimeDecoder("OffsetTime", OffsetTime.parse)
+  implicit val zonedDateTimeDecoder: CellDecoder[ZonedDateTime] = javaTimeDecoder("ZonedDateTime", ZonedDateTime.parse)
+
+  private def javaTimeDecoder[T](name: String, t: String => T): CellDecoder[T] = s =>
+    Either.catchOnly[DateTimeParseException](t(s))
+      .leftMap(t => new DecoderError(s"Couldn't parse $name: ${t.getMessage}", t))
```

Comment:
No need to repeat the t message in the wrapping error. Also convention I used in exception messages is to start with lower case.